datus/cli/generation_hooks.py

```
@@ -245,6 +397,60 @@ async def _handle_sql_summary_result(self, result):
+
+            # Print completion separator to prevent action stream from overwriting
+            self.console.print("\n" + "=" * 80)
+            self.console.print("[bold green]✓ Generation workflow completed, generating report...[/]", justify="center")
+            self.console.print("=" * 80 + "\n")
+
+            # Add delay to ensure message is visible before any new output
+            await asyncio.sleep(0.1)
+
+        except (KeyboardInterrupt, EOFError):
+            self.console.print("\n[yellow]✗ Sync cancelled by user[/]")
+            raise GenerationCancelledException("User interrupted")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)